ops/services/alerts/app_service_metrics/main.tf

```diff
@@ -120,47 +124,50 @@ requests
   }
 }
 
-resource "azurerm_monitor_metric_alert" "http_5xx_errors" {
-  name                = "${var.env}-api-5xx-errors"
-  description         = "${local.env_title} HTTP Server 5xx Errors >= 10"
+resource "azurerm_monitor_scheduled_query_rules_alert" "http_4xx_errors" {
+  name                = "${var.env}-api-4xx-errors"
+  description         = "${local.env_title} HTTP Server 4xx Errors (excluding 401s) >= 10"
+  location            = data.azurerm_resource_group.app.location
   resource_group_name = var.rg_name
-  scopes              = [var.app_service_id]
-  frequency           = "PT1M"
-  window_size         = "PT5M"
   severity            = var.severity
-  enabled             = contains(var.disabled_alerts, "http_5xx_errors") ? false : true
+  frequency           = 5
+  time_window         = 5
+  enabled             = contains(var.disabled_alerts, "http_4xx_errors") ? false : true
 
-  criteria {
-    aggregation      = "Count"
-    metric_name      = "Http5xx"
-    metric_namespace = "Microsoft.Web/sites"
-    operator         = "GreaterThanOrEqual"
-    threshold        = 10
+  data_source_id = var.app_insights_id
+
+  query = <<-QUERY
+requests
+${local.skip_on_weekends}
+| where toint(resultCode) >= 400 and toint(resultCode) < 500 and toint(resultCode) != 401 and timestamp >= ago(5m)
```

Comment:
We may also be able to skip 400s - we're using BadRequestException in a few different places now for user input errors (like a bad security passcode on account creation) and I don't foresee a situation where we'd want to alert on that behavior.